File: src/main/java/model/equipment/weapon/Spear.java
package model.equipment.weapon;

import model.support.allEnums.DmgType;
import model.support.allEnums.WeaponType;

public class Spear extends Weapon {
    public Spear(double weight, int quality) {
        super(WeaponType.SPEAR, weight, quality);
        setDamage();
    }

    private void setDamage() {
        damage.put(DmgType.PIERCING, quality);
    }

}

File: src/main/java/model/equipment/weapon/Spear.kt
package model.equipment.weapon

import model.support.enums.Skill.SPEAR_SKILL
import model.support.enums.Specification.*
import model.support.enums.WeaponType.SPEAR

class Spear(weight: Double, quality: Int) : Weapon(SPEAR, SPEAR_SKILL, weight, quality, false) {
    init {
        damage = mapOf(
                BLUNT to 0.0,
                CUT to 0.0,
                SLASH to 0.0,
                PIERCING to quality.toDouble(),
                CRUSH to 0.0)
    }
}
